homeassistant/helpers/issue_registry.py

```diff
@@ -132,7 +132,7 @@ def async_get_or_create(
         translation_placeholders: dict[str, str] | None = None,
     ) -> IssueEntry:
         """Get issue. Create if it doesn't exist."""
-
+        self.hass.verify_event_loop_thread("async_get_or_create")
```

Comment:
Once all the checks are relocated, the plan is to improve the messaging and than add some dev docs to link to that tells the dev what to do instead